### PR TITLE
Add dummy letter to upload widget ID for interfaces starting with a number

### LINKED ISSUE #3256: Unsupported upload interface type breaks later file uploads widgets
I have a reader study with a 3d-liver-model interface (.obj kind). The following likely also extends to archive items.

We don't support uploading these via the UI.

However, when one of these interfaces is present all subsequent file upload widgets are not loaded on page: https://grand-challenge.org/reader-studies/chris-test-reader-study-the-second-coming/display-sets/create-single/

image

I've since removed the display set with the interface to quickly do a workaround.

Not sure if this needs to be fixed or is so corner case that we can safely ignore it.

### PR DESCRIPTION
**Pull request by amickan**
An id name cannot start with a number in CSS (despite being valid HTML). Interfaces starting with a number break our upload widget because of that. 

Closes #3256

### CODE REVIEW

**Comment by amickan** [comment]
I just realized while testing that this is also an issue for htmx and thus for the FlexibleImageWidget We currently don't have any image interfaces that start with a number, so it's not a problem yet. I'm wondering if instead of adding a similar workaround for the htmx views where necessary, it might be better to addd a restriction for interface names?

File: app/grandchallenge/uploads/widgets.py [hunk 1]
```diff
@@ -11,8 +13,12 @@ def __init__(self, *args, allowed_file_types=None, **kwargs):
 
     def get_context(self, *args, **kwargs):
         context = super().get_context(*args, **kwargs)
+        widget_id = context["widget"]["attrs"]["id"]
+        if re.match(r"^\d", widget_id):
+            # id names cannot start with a number in CSS
```
Comment (jmsmkn) [inline app/grandchallenge/uploads/widgets.py:18]: Why not prefix everything for consistency? Or just flip it around to `allowedFileTypes{context['widget']['attrs']['id']`?

**Comment by jmsmkn** [comment]
> I just realized while testing that this is also an issue for htmx and thus for the FlexibleImageWidget We currently don't have any image interfaces that start with a number, so it's not a problem yet. I'm wondering if instead of adding a similar workaround for the htmx views where necessary, it might be better to addd a restriction for interface names?

I think fix the frontend instead, would be a strange restriction.

File: app/grandchallenge/uploads/widgets.py (hunk 1 above)
Comment (amickan) [reply app/grandchallenge/uploads/widgets.py]: True, prefixing all makes sense.

**Review by jmsmkn — APPROVED** [review]
